pkg/executor/builder.go

```diff
@@ -2743,6 +2749,9 @@ func (b *executorBuilder) buildAnalyzeColumnsPushdown(
 		startTS = uint64(val.(int))
 	})
 	concurrency := b.ctx.GetSessionVars().AnalyzeDistSQLScanConcurrency()
+	if concurrency <= 0 {
+		concurrency = adaptiveAnlayzeDistSQLConcurrency(context.Background(), b.ctx)
+	}
```

Comment:
Could we merge this logic with `AnalyzeDistSQLScanConcurrency` or create a new function, since they are always called together?

Reply:
Done, it has been merged.